app/src/main/java/com/futurice/freesound/feature/search/DefaultSearchDataModel.java

```diff
@@ -59,6 +62,7 @@
 
     @Override
     public Completable querySearch(@NonNull final String query) {
+        searchStateOnceAndStream.onNext(SEARCH_TRIGGERED());
```

Comment:
Make this a side-effect of subscribing to the returned `Completable` - otherwise progress will begin as soon as this is called, not when the actual action is performed.